sale_order_invoice_amount/models/sale_order.py

```diff
@@ -26,42 +28,51 @@ class SaleOrder(models.Model):
         "amount_total",
         "invoice_ids.state",
     )
-    def _compute_invoice_amount(self):
-        for rec in self:
-            if rec.state != "cancel" and rec.invoice_ids:
-                rec.invoiced_amount = 0.0
-                for invoice in rec.invoice_ids:
-                    if invoice.state != "cancel":
-                        if (
-                            invoice.currency_id != rec.currency_id
-                            and rec.currency_id != invoice.company_currency_id
-                        ):
-                            rec.invoiced_amount += invoice.currency_id._convert(
-                                invoice.amount_total_in_currency_signed,
-                                rec.currency_id,
-                                invoice.company_id,
-                                invoice.invoice_date or fields.Date.today(),
+    def _compute_amount_invoiced(self):
+        if not self.env.company.enable_amount_invoiced_based_on_quantity:
+            return super()._compute_amount_invoiced()
+        else:
+            for rec in self:
+                if rec.state != "cancel" and rec.invoice_ids:
+                    rec.amount_invoiced = 0.0
```

Comment:
Isn't it possible to affect zero value in a missing case than every time (and then affect another value)?

Reply:
You are right, changed!